share/functions/__fish_print_interfaces.fish

```diff
@@ -3,6 +3,11 @@ function __fish_print_interfaces --description "Print a list of known network in
         set -l interfaces /sys/class/net/*
         string replace /sys/class/net/ '' $interfaces
     else # OSX/BSD
-        command ifconfig -l | string split ' '
+        set -l os (uname)
+        if string match -q "OpenBSD" -- $os
+            command ifconfig | grep -e "^[a-z]" | awk -F: '{print $1}' | string split ' '
```

Comment:
Again, the grep can be trivially replaced with `string match -r "^[a-z]"`. The `awk` should be `string replace -r ':.*' ''`.

Reply:
looks like the `string match -r "^[a-z]"` part only returns the first character of each match.. is there a way to make it print the entire line when the RE is matched?

Reply:
You need to either make the regex match the entire line (`"^[a-z]+$"`) or use the "-e"/"--entire" flag.

Sorry, I forgot about that!